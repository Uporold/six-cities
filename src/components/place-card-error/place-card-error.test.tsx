import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import PlaceCardError from "./place-card-error";

const mockStore = configureStore([]);

it(`PlaceCardError render correctly`, () => {
  const store = mockStore({
    DATA: {
      errorHotelIds: [0, 1, 2, 3],
      errorMessage: `Test error`,
    },
    USER: {
      authorizationStatus: true,
    },
  });
  const tree = renderer
    .create(
      <Provider store={store}>
        <PlaceCardError hotelId={0} />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
