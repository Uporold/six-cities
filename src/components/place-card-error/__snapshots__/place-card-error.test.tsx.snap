// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCardError render correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "background": "rgba(50,50,50,0.6)",
      "color": "white",
      "cursor": "pointer",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "left": "0",
      "position": "absolute",
      "top": "0",
      "width": "100%",
      "zIndex": "999",
    }
  }
>
  <p>
    Test error
  </p>
  <p>
    Click to close
  </p>
</div>
`;
